app/com/arpnetworking/metrics/portal/reports/impl/DatabaseReportExecutionRepository.java

```diff
@@ -0,0 +1,311 @@
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.portal.reports.impl;
+
+import com.arpnetworking.metrics.portal.scheduling.JobExecutionRepository;
+import com.arpnetworking.steno.Logger;
+import com.arpnetworking.steno.LoggerFactory;
+import com.google.common.base.Throwables;
+import io.ebean.EbeanServer;
+import io.ebean.Transaction;
+import models.ebean.ReportExecution;
+import models.internal.Organization;
+import models.internal.reports.Report;
+import models.internal.scheduling.JobExecution;
+
+import java.time.Instant;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicBoolean;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.persistence.EntityNotFoundException;
+import javax.persistence.PersistenceException;
+
+/**
+ * Implementation of {@link JobExecutionRepository} for {@link Report} jobs using a SQL database.
+ *
+ * @author Christian Briones (cbriones at dropbox dot com)
+ */
+public final class DatabaseReportExecutionRepository implements JobExecutionRepository<Report.Result> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DatabaseReportExecutionRepository.class);
+
+    private final AtomicBoolean _isOpen = new AtomicBoolean(false);
+    private final EbeanServer _ebeanServer;
+
+    /**
+     * Public constructor.
+     *
+     * @param ebeanServer Play's {@code EbeanServer} for this repository.
+     */
+    @Inject
+    public DatabaseReportExecutionRepository(@Named("metrics_portal") final EbeanServer ebeanServer) {
+        _ebeanServer = ebeanServer;
+    }
+
+    @Override
+    public void open() {
+        assertIsOpen(false);
+        LOGGER.debug().setMessage("Opening DatabaseReportExecutionRepository").log();
+        _isOpen.set(true);
+    }
+
+    @Override
+    public void close() {
+        assertIsOpen();
+        LOGGER.debug().setMessage("Closing DatabaseReportExecutionRepository").log();
+        _isOpen.set(false);
+    }
+
+
+    /**
+     * Get the most recently scheduled execution, if any.
+     * <p>
+     * This could possibly return an execution that's pending completion.
+     *
+     * @param jobId The UUID of the job that completed.
+     * @param organization The organization owning the job.
+     * @return The last successful execution.
+     * @throws NoSuchElementException if no job has the given UUID.
+     */
+    public Optional<JobExecution<Report.Result>> getLastScheduled(final UUID jobId, final Organization organization)
+            throws NoSuchElementException {
+        assertIsOpen();
+        return _ebeanServer.find(ReportExecution.class)
+                .orderBy()
+                .desc("scheduled")
+                .where()
+                .eq("report.uuid", jobId)
+                .eq("report.organization.uuid", organization.getId())
+                .setMaxRows(1)
```

Comment:
Yes but I'm not sure if there's an alternative here. I want the most recently scheduled and since there are many executions for (most) jobs then removing the maxRows would cause it to throw most of the time.

Reply:
You could set it to 2.

Reply:
Oh! Nevermind.